tfs/hdf.py

```diff
@@ -0,0 +1,93 @@
+"""
+HDF5 Tools
+----------
+
+Additional tools for reading and writing TfsDataFrames into hdf5 files.
+"""
+import pandas as pd
+from pathlib import Path
+
+from typing import Union
+from tfs import TfsDataFrame
+
+try:
+    import h5py
+except ImportError:
+    h5py = None
+
+try:
+    import tables
+except ImportError:
+    tables = None
+
+import logging
+
+LOGGER = logging.getLogger(__name__)
+
+
+def write_hdf(path: Union[Path, str], df: TfsDataFrame, **kwargs):
+    """Write TfsDataFrame to hdf5 file. The dataframe will be written into
+    the group ``data``, the headers into the group ``headers``.
+    Only one frame per file is allowed.
+
+    Args:
+        path (Path, str): Path of the output file.
+        df (TfsDataFrame): TfsDataFrame to write.
+        kwargs: kwargs to be passed to pandas ``DataFrame.to_hdf()``.
+                ``key`` is not allowed and ``mode`` needs to be ``w`` if the
+                output file already exists.
+    """
+    _check_imports()
+    # Check for `key` kwarg (forbidden) ---
+    if "key" in kwargs:
+        raise AttributeError("The argument 'key' is not allowed here, "
+                             "as only one TfsDataFrame per file is supported.")
+
+    # Check for `mode` kwarg (allowed under circumstances but generally ignored) ---
+    user_mode = kwargs.pop('mode', None)
+    if user_mode is not None and user_mode != "w":
```

Comment:
I guess this should also check that the file already exists to be consistent with the docstring? Or should we modify the docstring to say "please don't provide mode, it has to be `w` anyway (which we hard-code just below)"?

Reply:
It checks that the file exist in the line below and raises an error if it does